### DISCUSSION TITLE
hide cast option on npc/player (pvm)

### DISCUSSION THREAD

**Discussion by creilson** [runelite/runelite#15123] [Ideas]
in response to [updated third party client guidelines](https://secure.runescape.com/m=news/third-party-client-guidelines?oldschool=1), would it be possible to hide cast options on npcs or players OUTSIDE OF PVP?

i don't think this would go against the guideline:
<html>
<body>

Removes or deprioritises attack, cast (or similar) options from the minimenu **in PVP** | Combat
-- | --

</body>
</html>

i think it would be helpful for avoiding casting on thralls or other players in pvm. thanks.

**Comment by RedCondor6** [comment]
+1

**Comment by LlemonDuck** [comment]
This could still be considered disallowed under
> Makes it easier to target 3D entities with a spell by removing some options

**Reply by creilson** [reply]
yeah might be worth getting clarification. the way i interpreted that is removing options on the entity you're meaning to cast on.

**Reply by creilson** [reply]
if applicable, another way to implement could be deprioritizing this option (lowering in stack).

**Reply by leadanymind** [reply]
I agree with @creilson here. It should be allowed as an option, since it's almost certainly not an intended mechanic in the code. It's clearly a bug, since it's not possible to do that in tob for example. I think the error message in the chat window is a "quick fix" made on Jagex part, but _should_ be toggleable at the very least.
